arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Option.kt

```diff
@@ -1029,6 +1255,14 @@ public inline fun <A> Option<A>.ensure(error: () -> Unit, predicate: (A) -> Bool
 /**
  * Returns an Option containing all elements that are instances of specified type parameter [B].
  */
+@Deprecated(
+  NicheAPI + "Prefer using option DSL or flatMap",
```

Comment:
This aligns with stdlib for list.filterIsInstance. let's not deprecate this.

Reply:
Is this an API that is commonly used in your projects @myuwono? It's an API I personally find kind-of strange on `Option`, even on `List` I think it's not commonly used 🤔

I am having a hard time thinking of a good use-case. I hope that `Option<Any>` is an extremely rare case 😁
Otherwise inheritance rules should apply? 🤔

Reply:
Similarly to Lists, this is useful for a call that returns a sealed type @nomisRev. For instance if a database call returns `Option<Document>` where document is a sealed class with many subtypes, and we're in are interested in only one of them e.g. `Option<Document.ExternalMetadata>`.

Reply:
Right, you need it for `narrow`'ing. So an `Option` alternative to `as?`.
Since this only makes sense for `Option`, lets keep it 👍